code/modules/mob/mob.dm

```diff
@@ -578,9 +578,15 @@ var/list/slot_equipment_priority = list( \
 //puts the item "W" into an appropriate slot in a human's inventory
 //returns 0 if it cannot, 1 if successful
 /mob/proc/equip_to_appropriate_slot(obj/item/W)
-	if(!istype(W)) return 0
+	if(!istype(W))
+		return 0
 
 	for(var/slot in slot_equipment_priority)
+		if(get_item_by_slot(slot))
```

Comment:
You shouldn't really have to call this twice.

    var/obj/item/S = get_item_by_slot
    if(S)